pkg/group/repository.go

```diff
@@ -23,13 +24,32 @@ func NewRepository(db *gorm.DB) *repository {
 func (r repository) find(name string) (*model.Group, error) {
 	var group *model.Group
 	err := r.db.
-		Preload("ClusterConfiguration").
+		Joins("ClusterConfiguration").
 		Where("name = ?", name).
 		First(&group).Error
 	if errors.Is(err, gorm.ErrRecordNotFound) {
 		return nil, errdef.NewNotFound("group %q doesn't exist", name)
 	}
 
+	if err != nil {
+		return nil, fmt.Errorf("failed to find group with details: %v", err)
```

Comment:
Is this `find(name)` method also getting the group with details? Or should this error message go into `findWithDetails(name string)` 🤔